src/test/java/be/seeseemelk/mockbukkit/TestPlugin.java

```diff
@@ -126,6 +130,24 @@ public void onAsyncChat(AsyncPlayerChatEvent event)
 		}
 	}
 
+	@EventHandler
+	public void onAsyncPreLogin(AsyncPlayerPreLoginEvent event)
+	{
+		if (denyAsyncPreLoginEvent)
+		{
+			event.setLoginResult(AsyncPlayerPreLoginEvent.Result.KICK_OTHER);
+		}
+	}
+
+	@EventHandler
+	public void onLogin(PlayerLoginEvent event)
+	{
+		if (denyLoginEvent)
+		{
+			event.setResult(PlayerLoginEvent.Result.KICK_OTHER);
+		}
+	}
+
```

Comment:
I don't think this needs to be added, as it's a bit of a niche use case. This is just for common stuff that people will test.